### PR TITLE
Allow ICSP resource creation

### PR DESCRIPTION
**Pull request by omrirh**

### CODE REVIEW

File: ocp_resources/image_content_source_policy.py [hunk 1]
```diff
@@ -7,3 +7,28 @@ class ImageContentSourcePolicy(Resource):
...
     api_group = Resource.ApiGroup.OPERATOR_OPENSHIFT_IO
+
+    def __init_(self, source=None, mirrors=None, **kwargs):
+        """
+        Create/Manage ICSP configuration object
+
+        Args:
+            source (string): the repository that users refer to, e.g. in image pull specifications.
+            mirrors (list): one or more repositories that may also contain the same images.
+                The order of mirrors in this list is treated as the user’s desired priority
```
Comment (rnetser) [inline ocp_resources/image_content_source_policy.py:18]: this allows adding only one policy; but there could be a few.
should get repositoryDigestMirrors (repository_digest_mirrors) which will get a list of dicts.

**Review by rnetser — CHANGES_REQUESTED** [review]
- Please follow https://github.com/RedHatQE/openshift-python-wrapper/blob/main/CONTRIBUTING.md; a reference to the api is missing
- Please update PR title; the addition is to allow resource creation not update.

File: ocp_resources/image_content_source_policy.py (hunk 1 above)
Comment (omrirh) [reply ocp_resources/image_content_source_policy.py:20]: done; let me know wyt

File: ocp_resources/image_content_source_policy.py [hunk 2]
```diff
@@ -7,3 +7,24 @@ class ImageContentSourcePolicy(Resource):
...
     api_group = Resource.ApiGroup.OPERATOR_OPENSHIFT_IO
+
+    def __init_(self, repository_digest_mirrors=None, **kwargs):
+        """
+        Create/Manage ICSP configuration object. API reference:
+        https://docs.openshift.com/container-platform/4.12/rest_api/operator_apis/imagecontentsourcepolicy-operator-openshift-io-v1alpha1.html
+
+        Args:
+            repository_digest_mirrors (list): [{<source>: <mirrors>}, ..., {<source>: <mirrors>}]
+            source (string): the repository that users refer to, e.g. in image pull specifications.
```
Comment (myakove) [inline ocp_resources/image_content_source_policy.py:18]: remove `source` and `mirrors` from the docstring.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/image_content_source_policy.py (hunk 2 above)
Comment (omrirh) [reply ocp_resources/image_content_source_policy.py:18]: shouldn't there be a doc for what to pass inside `repository_digest_mirrors` ?

File: ocp_resources/image_content_source_policy.py [hunk 3]
```diff
@@ -7,3 +7,21 @@ class ImageContentSourcePolicy(Resource):
...
 
     api_group = Resource.ApiGroup.OPERATOR_OPENSHIFT_IO
+
+    def __init__(self, repository_digest_mirrors=None, **kwargs):
+        """
+        Create/Manage ICSP configuration object. API reference:
+        https://docs.openshift.com/container-platform/4.12/rest_api/operator_apis/imagecontentsourcepolicy-operator-openshift-io-v1alpha1.html
+
+        Args:
+            repository_digest_mirrors (list): [{<source>: <mirrors>}, ..., {<source>: <mirrors>}]
```
Comment (rnetser) [inline ocp_resources/image_content_source_policy.py:17]: - `repository_digest_mirrors` is mandatory, please raise if not passed
- please explain what is the format of `repository_digest_mirrors` - should be a list of dictionaries, each containing `mirrors` key with a list of mirrors and a `source` key which contains a source str

```
[{'mirrors': ['mirror1',],  'source': 'src'}]

```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/image_content_source_policy.py (hunk 3 above)
Comment (omrirh) [reply ocp_resources/image_content_source_policy.py]: done

**Review by myakove — APPROVED** [review]

File: ocp_resources/image_content_source_policy.py [hunk 4]
```diff
@@ -7,3 +7,27 @@ class ImageContentSourcePolicy(Resource):
...
+        """
+        Create/Manage ICSP configuration object. API reference:
+        https://docs.openshift.com/container-platform/4.12/rest_api/operator_apis/imagecontentsourcepolicy-operator-openshift-io-v1alpha1.html
+
+        Args:
+            repository_digest_mirrors (list of dict): e.g. [{<source>: <mirrors>}, ..., {<source>: <mirrors>}]
+            where for each dict:
+                - <source> (str) is the repository that users refer to, e.g. in image pull specifications
+                - mirrors (list) is one or more repositories (str) that may also contain the same images. The order
+                of mirrors in this list is treated as the user’s desired priority
```
Comment (rnetser) [inline ocp_resources/image_content_source_policy.py:21]:
```suggestion
            repository_digest_mirrors (list of dict): e.g. [{source: <str>, mirrors: <list>}, ..., {source: <str>, mirrors: <list>}]
            - source - the repository that users refer to, e.g. in image pull specifications
            - mirrors - one or more repositories (str) that may also contain the same images. The order
                of mirrors in this list is treated as the user’s desired priority
```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/image_content_source_policy.py (hunk 4 above)
Comment (omrirh) [reply ocp_resources/image_content_source_policy.py]: done

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**